Traktor: Implemented autocomplete for SyntaxRichEdit

jayrulez opened the pull request:
https://github.com/apistol78/traktor/commit/1157c2cea76d2b1608a34dc4894ca3d26c4a580d

For this specific change, I don't really like the approach of using a flag.

However, I couldn't find a good way to consume tab and return from the autocomplete popup.
If you have a suggestion, then please let me know.

Also, maybe LuaAutocompleteProvider should be moved to the Script editor plugin?
I intend to expand this to provide autocomplete from the class registry later.

jayrulez committed:
Autocomplete - Add source folder to solution

jayrulez committed:
Traktor: Added autocomplete feature for SyntaxRichEdit

jayrulez committed:
Traktor: Consume tab and new line KeyEvent when autocomplete is used

jayrulez committed:
Cleaned up duplicated code

jayrulez committed:
Added Autocomplete dir to Linux solution

jayrulez committed:
Moved Autocomplete to SyntaxRichEdit filter

jayrulez committed:
Reorganized files

apistol78 commented:
Really good work!
I've started looking through it now, and some minor thoughts

- Do you think we should extend the ScriptOutlineLua class to provide data to the provider?
- The suggestion widget should probably be a ToolForm widget.
- Using AutoWidget as a basis for the list of suggestion would automate scroll bar functionality.

jayrulez committed:
Made AutocompletePopup a ToolForm widget

Use AutoWidget as the basis for autocomplete list. This automates scroll functionality

jayrulez committed:
Renamed Lua autocomplete provider

jayrulez committed:
Populate autocomplete from script outline (WIP)

apistol78 committed:
Merge branch 'master' into autocomplete2
